src/identity/quartzOrganizations/actions/thunks/index.ts

```diff
@@ -33,3 +41,20 @@ export const getQuartzOrganizationsThunk = () => async (
     dispatch(notify(updateQuartzOrganizationsFailed()))
   }
 }
+
+export const updateDefaultOrgThunk = (
+  oldDefaultOrg: DefaultOrg,
+  newDefaultOrg: DefaultOrg
+) => async (dispatch: Dispatch<Actions>) => {
+  try {
+    dispatch(setQuartzOrganizationsStatus(RemoteDataState.Loading))
+    await putDefaultQuartzOrg(newDefaultOrg.id)
+    dispatch(setQuartzDefaultOrg(oldDefaultOrg.id, newDefaultOrg.id))
+
+    dispatch(setQuartzOrganizationsStatus(RemoteDataState.Done))
+    dispatch(notify(accountDefaultSettingSuccess(newDefaultOrg.name)))
+  } catch (err) {
+    dispatch(setQuartzOrganizationsStatus(RemoteDataState.Error))
+    dispatch(notify(accountDefaultSettingError(newDefaultOrg.name)))
```

Comment:
This doesn't need to be changed right now, but I think we should consider not notifying from the thunks layer if we're going to be setting the `RemoteDataState` to `Error`. Dispatching actions for notifications in the thunk layer will make re-usability of the thunks a little bit harder because it tightly couples the error handling with the messaging. It seems like it should be the Component layer's responsibility to notify the user based on whether the `RemoteDataState` is set to `Done` or `Error`.

Give it a think and let me know how you feel

Reply:
@hoorayimhelping Thanks - completely agree on placing the notifications in the Component layer instead. I'll save the notification for the UserProfile component itself. Some thoughts here:

On using `RemoteDataState`, I'm sort of divided. Checking `if (status === 'done')` versus `if (status === 'error')` as the condition for deciding which notification we dispatch doesn't seem ideal in my view, because either we either:

1. have to create a separate 'status' in redux for every new API request or thunk we create (updateDefaultOrgStatus, getOrgsStatus, getBillingInfoStatus) - which is annoying and cluttered, or 
2. have to uncross the wires from multiple thunks potentially updating the same redux `status` (e.g., API requests that are getting orgs or setting new org properties both set the `status` for `organizations` to 'error' if they go wrong, so our `if` block can't distinguish the error associated with one failed api request from another). That seems worse to me because then we know the app is in an error state but we can't trace why, and there's a risk the user gets an error notification for the wrong reason.

Maybe we throw an error in the thunk if the API request fails; and in the UI, we keep the existing `try/catch` blocks, using them to dispatch the notification so that it's tied directly to the error, instead of indirectly to the change in the redux state. We can also update the `RemoteDataState`, but maybe this is a bigger question about what those data state updates are being used as an input for.

Reply:
>Maybe we throw an error in the thunk if the API request fails; and in the UI, we keep the existing try/catch blocks, using them to dispatch the notification so that it's tied directly to the error, instead of indirectly to the change in the redux state.

hahaha, we've come full-circle on the conversation about this we had a couple of weeks ago when you asked if we should return the error from the thunk, or dispatch an action. I think you make good points in both cases, but I tend to think we should handle the returned error directly, for the reasons you outlined here.

Reply:
Hah, agreed. There's probably a bigger conversation to be had about the use of the redux status - it's possible I'm missing the utility of doing it that way, but we can always reconsider if this behavior doesn't seem optimal.